kalite/updates/api_views.py

```diff
@@ -235,99 +203,14 @@ def delete_language_pack(request):
-            child = annotate_topic_tree(child_node, level=level + 1, statusdict=statusdict, lang_code=lang_code)
-            if not child:
-                continue
-            elif child["extraClasses"] == "unstarted":
-                complete = False
-            elif child["extraClasses"] == "partial":
-                complete = False
-                unstarted = False
-            elif child["extraClasses"] == "complete":
-                unstarted = False
-            children.append(child)
-
-        if not children:
-            # All children were eliminated; so eliminate self.
-            return None
-
-        return {
-            "title": _(node["title"]),
-            "tooltip": re.sub(r'<[^>]*?>', '', _(node.get("description")) or ""),
-            "folder": True,
-            "key": node["id"],
-            "children": children,
-            "extraClasses": complete and "complete" or unstarted and "unstarted" or "partial",
-            "expanded": level < 1,
-        }
-
-    elif node["kind"] == "Video":
-        video_id = node.get("youtube_id", node.get("id"))
-        youtube_id = get_youtube_id(video_id, lang_code=lang_code)
-
-        if not youtube_id:
-            # This video doesn't exist in this language, so remove from the topic tree.
-            return None
-
-        # statusdict contains an item for each video registered in the database
-        # will be {} (empty dict) if there are no videos downloaded yet
-        percent = statusdict.get(youtube_id, 0)
-        vid_size = None
-        status = None
-
-        if not percent:
-            status = "unstarted"
-            vid_size = get_remote_video_size(youtube_id) / float(2 ** 20)  # express in MB
-        elif percent == 100:
-            status = "complete"
-            vid_size = get_local_video_size(youtube_id, 0) / float(2 ** 20)  # express in MB
-        else:
-            status = "partial"
-
-        return {
-            "title": _(node["title"]),
-            "tooltip": re.sub(r'<[^>]*?>', '', _(node.get("description")) or ""),
-            "key": youtube_id,
-            "extraClasses": status,
-            "size": vid_size,
-        }
-
-    return None
-
-
 @require_admin
 @api_handle_error_with_json
-def get_annotated_topic_tree(request, lang_code=None):
-    call_command("videoscan")  # Could potentially be very slow, blocking request... but at least it's via an API request!
+def get_update_topic_tree(request, lang_code=None):
 
+    parent = request.GET.get("parent")
     lang_code = lang_code or request.language      # Get annotations for the current language.
-    statusdict = dict(VideoFile.objects.values_list("youtube_id", "percent_complete"))
 
-    return JsonResponse(annotate_topic_tree(softload_json(TOPICS_FILEPATHS.get(CHANNEL), logger=logging.debug, raises=False), statusdict=statusdict, lang_code=lang_code))
+    return JsonResponse(get_topic_update_nodes(parent=parent, language=lang_code))
```

Comment:
This returns layer by layer, so we can lazily load the topic tree into the front end.